horovod/common/wire/message.fbs

```diff
@@ -28,7 +28,8 @@ enum DataType:byte {
     HOROVOD_FLOAT16 = 6,
     HOROVOD_FLOAT32 = 7,
     HOROVOD_FLOAT64 = 8,
-    HOROVOD_BOOL = 9
+    HOROVOD_BOOL = 9,
+    HOROVOD_BYTE = 10
```

Comment:
I'm a little confused by `HOROVOD_BYTE`.  In what cases would it be used in place of `HOROVOD_UINT8`?

Reply:
I was under the impression that we needed to keep the enums consistent between `enum DataType` in [message.h](https://github.com/horovod/horovod/blob/master/horovod/common/message.h) and `enum DataType` in [message.fbs](https://github.com/horovod/horovod/blob/master/horovod/common/wire/message.fbs)/[message_generated.h](https://github.com/horovod/horovod/blob/master/horovod/common/wire/message_generated.h). Am i incorrect? `HOROVOD_BYTE` was already in [message.h](https://github.com/horovod/horovod/blob/master/horovod/common/message.h).

Reply:
Ah, I see what happened.  The `HOROVOD_BYTE` dtype was added in https://github.com/horovod/horovod/commit/aa605d6de1e07d1184e8793b6d4f65cee9d4efb3 to abstract an API call to `MPI_Bcast`, but was subsequently simplified in https://github.com/horovod/horovod/commit/669eb9e0dde25f7475e6a31630ae4b9a704ae106.

The result is that `HOROVOD_BYTE` is no longer used for anything, so I think we can safely remove it.